### PR TITLE
Test Adapter Fixes

### PR DESCRIPTION
**Pull request by adamdriscoll**
I removed PSate support and simplified the code for Pester. It doesn't do any version checking and works with the most recent version of Pester. I am getting the results from the pipeline rather than an XML file. I also fixed the issue where tests that weren't part of a context would not be found.

### CODE REVIEW

**Comment by adamdriscoll** [comment]
@AndreSayreMSFT  Can you try this build for the tests you were running for this issue? https://github.com/Microsoft/poshtools/issues/439

**Comment by adamdriscoll** [comment]
Fixes #228

File: PowerShellTools.TestAdapter/PowerShellTestContainerDiscoverer.cs [hunk 1]
```diff
@@ -230,7 +230,7 @@ where IsTestFile(item)
 
         private static bool IsPs1File(string path)
```
Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestContainerDiscoverer.cs]: IsTestsPs1File?

Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestContainerDiscoverer.cs]: Or IsPowerShellTestFile

File: PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs [hunk 2]
```diff
@@ -71,62 +52,77 @@ public static List<TestCase> GetTests(IEnumerable<string> sources, ITestCaseDisc
...
-                            SendMessage(TestMessageLevel.Informational, "Test name was empty. Skipping test.", logger);
-                            continue;
-                        }
+                foreach(var tag in tags)
+                {
+                    testcase.Traits.Add(tag, string.Empty);
+                }
+                
+                SendMessage(TestMessageLevel.Informational,
+                    String.Format("Adding test [{0}] in {1} at {2}.", describeName, source, testcase.LineNumber), logger);
```
Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs]: nit: string vs String

Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs]: Do we want these user-facing strings to be in a Resources file?

File: PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs [hunk 3]
```diff
@@ -71,62 +52,77 @@ public static List<TestCase> GetTests(IEnumerable<string> sources, ITestCaseDisc
...
 
-                        var displayName = String.Format("{0} It {1}", contextName, itName);
-                        var fullName = String.Format("{0}||{1}||{2}||{3}", "Pester", describeName, contextName, itName);
+                if (discoverySink != null)
+                {
+                    SendMessage(TestMessageLevel.Informational, "Sending test to sync.", logger);
```
Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs]: User facing string in res file?

File: PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs [hunk 4]
```diff
@@ -71,62 +52,77 @@ public static List<TestCase> GetTests(IEnumerable<string> sources, ITestCaseDisc
...
+                */
+            }
+        }
+
+        private static string GetParentContextName(IMessageLogger logger, Ast ast)
+        {
+            if  (ast.Parent is CommandAst &&
+                (ast.Parent as CommandAst).GetCommandName() != null &&
+                (ast.Parent as CommandAst).GetCommandName().Equals("context", StringComparison.OrdinalIgnoreCase))
+            {
```
Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs]: if( ast.parent is commandAst && string.Equals( "context", ((CommandAst)ast.Parent).GetCommandName(), stringcomparison.ordinalignoreCase) 

?

File: PowerShellTools.TestAdapter/PowerShellTestExecutor.cs [hunk 5]
```diff
@@ -127,13 +97,111 @@ public void Cancel()
...
-        public abstract PowerShellTestResult RunTest(PowerShell powerShell, TestCase testCase, IRunContext runContext);
+
+                    var context = result.Properties["Context"].Value as string;
+                    var name = result.Properties["Name"].Value as string;
+                    var stackTraceString = result.Properties["StackTrace"].Value as string;
+                    var errorString = result.Properties["FailureMessage"].Value as string;
+
+                    if (!string.IsNullOrEmpty(stackTraceString))
+                    {
+                        stackTrace.AppendFormat("{0} it {1}\r\n{2}\r\n\r\n", context, name, stackTraceString);
```
Comment (AndreSayreMSFT) [inline PowerShellTools.TestAdapter/PowerShellTestExecutor.cs]: Do we want to AppendLine as needed so we don't need to add in a bunch of the te \r\n's?  [Line endings may differ on environment]

**Comment by AndreSayreMSFT** [comment]
@adamdriscoll Now with this chunk of code, VS doesn't seem to be detecting any tests :(

**Comment by adamdriscoll** [comment]
@AndreSayreMSFT Are your tests in *.tests.ps1 files? Anything in the output pane?

**Comment by AndreSayreMSFT** [comment]
@adamdriscoll I feel kind of silly, they were all named *test.ps1 -- The Default module project creates .test.ps1 -- Did we want to update that?  Other than that, looks good!

**Comment by adamdriscoll** [comment]
@AndreSayreMSFT Yes, please! I had the exact same issue yesterday... ;)

**Comment by AndreSayreMSFT** [comment]
https://github.com/Microsoft/poshtools/pull/478

**Pull request merged**